pkg/virt-controller/services/template.go

```diff
@@ -168,6 +169,18 @@ func CPUModelLabelFromCPUModel(vmi *v1.VirtualMachineInstance) (label string, er
 	return
 }
 
+func CPUFeatureLabelsFromCPUFeatures(vmi *v1.VirtualMachineInstance) []string {
+	var labels []string
+	if vmi.Spec.Domain.CPU != nil && vmi.Spec.Domain.CPU.Features != nil {
+		if len(vmi.Spec.Domain.CPU.Features) > 0 {
```

Comment:
you don't have to have this check here. If vmi.Spec.Domain.CPU.Features has 0 length, then for loop won't start.